heimlig/src/client/api.rs

```diff
@@ -1,52 +1,48 @@
 use crate::common::jobs::{Request, Response};
-
-#[derive(Clone, Eq, PartialEq, Debug)]
-pub enum Error {
-    /// Attempted to push to a full queue.
-    QueueFull,
-}
-
-/// Sink where the requests to the Core can be pushed to
-pub trait RequestSink<'a> {
-    /// Send a [Request] to the client through this sink.
-    fn send(&mut self, request: Request<'a>) -> Result<(), Error>;
-    fn ready(&self) -> bool;
-}
+use crate::common::queues;
+use crate::common::queues::RequestSink;
 
 /// An interface to send [Request]s to the HSM core and receive [Response]es from it.
 pub struct Api<'a, Req: RequestSink<'a>, Resp: Iterator<Item = Response<'a>>> {
-    requests_sink: Req,
-    responses_source: Resp,
+    requests: Req,
+    responses: Resp,
+}
+
+#[derive(Copy, Clone, Eq, PartialEq, Debug)]
+pub enum Error {
+    Queue(queues::Error),
 }
```

Comment:
Do we see other errors being necessary here? If not we could do `pub use queues::Error` or so to re-export the error?